lib/ActionField.js

```diff
@@ -30,7 +30,7 @@ class ActionField extends Immutable.Record({
 			map.set('name', json.name || map.name);
 			map.set('title', json.title || map.title);
 			map.set('type', json.type || map.type);
-			map.set('value', json.value || map.value);
+			map.set('value', (json.value === false || json.value === 0) ? json.value : json.value || map.value);
```

Comment:
I like the "allow everything except `undefined`" approach, too.  It is pretty simple to reason about and unlikely to result in a lot of side effects.